### ISSUE TITLE
coda_expression_fuzzer: Direct-leak in coda_expression_lex

### ISSUE THREAD

**Issue by schwehr** [stcorp/coda#65]
```
==176737==ERROR: LeakSanitizer: detected memory leaks

Direct leak of 2 byte(s) in 1 object(s) allocated from:
    #1 0x55f3647914ba in coda_expression_lex ../coda/libcoda/coda-expr-tokenizer.l:137:48
    #2 0x55f3647889da in coda_expression_parse libcoda/coda-expr-parser.c:2289:16
    #3 0x55f36478ff2e in coda_expression_from_string ../coda/libcoda/coda-expr-parser.y:1049:9
    #4 0x55f3646e9239 in LLVMFuzzerTestOneInput third_party/stcorp_coda/fuzz/coda_expression_fuzzer.cc:23:3
```

The test case is just `@j`

I believe the `strdup` ptr in `yylval.stringval` should get freed in each of the uses of `INDEX_VAR` where `expression_new` returns `NULL`
```
{INDEX}                 {
                            yylval.stringval = strdup(yytext);
                            return INDEX_VAR;
                        }
```

I think this is the one being triggered here (I have not tested it to see):

```
    | INDEX_VAR {
            $$ = coda_expression_new(expr_index_var, $1, NULL, NULL, NULL, NULL);
            if ($$ == NULL) YYERROR;
        }
```

But there might be a better place where a free needs to happen on error.

```
grep INDEX_VAR *.[ly]
coda-expr-parser.y:%token    <stringval> INDEX_VAR
coda-expr-parser.y:    | INDEX_VAR
coda-expr-parser.y:    | FOR INDEX_VAR '=' intexpr TO intexpr DO voidexpr {
coda-expr-parser.y:    | FOR INDEX_VAR '=' intexpr TO intexpr STEP intexpr DO voidexpr {
coda-expr-parser.y:    | FUNC_WITH '(' INDEX_VAR '=' intexpr ',' voidexpr ')' {
coda-expr-parser.y:    | FUNC_WITH '(' INDEX_VAR '=' intexpr ',' boolexpr ')' {
coda-expr-parser.y:    | INDEX_VAR {
coda-expr-parser.y:    | FUNC_WITH '(' INDEX_VAR '=' intexpr ',' intexpr ')' {
coda-expr-parser.y:    | FUNC_WITH '(' INDEX_VAR '=' intexpr ',' floatexpr ')' {
coda-expr-parser.y:    | FUNC_WITH '(' INDEX_VAR '=' intexpr ',' stringexpr ')' {
coda-expr-tokenizer.l:                            return INDEX_VAR;
```

[testcase-5768301283639296.zip](https://github.com/stcorp/coda/files/4581169/testcase-5768301283639296.zip)

**Comment by svniemeijer** [comment]
Fixed in e7f9280e67bdbf23b1197c0a01822b9a6affaa58

**Comment by schwehr** [comment]
Verified
